### PR TITLE
add person info modal to people picker

### LINKED ISSUE #635: Show full person profile within modal
### Product reqs
- Clicking on the 'i' button within a person pod should open up a modal
- The lightbox should contain all of the information on a person collected from the eLife API
- The information should be separated into its component fields
- Clicking 'Add Editor' should close the lightbox and select the person (the same as hitting the '+' button without opening the modal
- Clicking 'Cancel' should close the modal with no changes to the UI
- ~Clicking 'Back' on the browser should close the modal~ - product req removed
- ~Clicking outside of the lightbox should close the modal~ - deprioritised & now captured as issue #1121 

### Designs
https://app.zeplin.io/project/5bacf042615a8b6e20e84a5f/screen/5bc48bf829495f5fbbfe62ed

screen shot 2018-11-12 at 16 19 06

### Tasks
- [x] Create PersonInfo component for use within ModalDialog
- [x] Refactor existing codebase to store research focuses & expertises separately from each other, so they can be displayed separately on the modal
- [x] Add FullPersonInfoModal into PersonPod - involves making it stateful, in order to keep track of whether modal is open/closed
- [x] Connect FullPersonInfoModal into EditorsPage (version with Accept button = "Remove Editor") & PeoplePickerControl (version with Accept button = "Add Editor")

### PR DESCRIPTION
**Pull request by jsms90**
#### What does this PR do?
- add new `PersonInfo` component - text that goes inside new modal dialog
- refactor existing `Paragraph` & `SmallParagraph` components to allow `secondary` prop for text colour (make consistent across the rest of the existing codebase)
- separate subjectAreas into focuses & expertises when making the API call to eLife's `/people` endpoint, so that we can separate these in the modal view
- refactor PersonPod to make it easier to add to
- render modal inside PersonPod, make PersonPod stateful to keep track of whether the modal is open
- hook up to PeoplePickerModal & PeoplePickerControl

To do:
- tests
- change `ModalDialog` in order to satisfy this product req:
> Clicking outside of the lightbox should close the modal

Could raise in a new ticket for the to dos?

#### Any relevant tickets

Closes #635 

#### How has this been tested?

Adjusted existing tests, but all testing for this new component TBD

- [ ] Unit / Integration tests
- [ ] Browser tests
- [ ] End2end tests

#### UI Changes

- [x] Has been reviewed against Designs
- [x] Necessary updates to styleguide

### CODE REVIEW

**Comment by jsms90** [comment]
"WIP" depending on whether we should complete the To Do list in this PR or raise new tickets?

The code itself & functionality can be reviewed now :smile:

**Comment by hdrury1** [comment]
Just taking a look on my local machine and noticed a couple of things that may be worth addressing now before the PR goes in:

screen shot 2018-11-27 at 11 25 59
screen shot 2018-11-27 at 11 30 49

- Text in the PersonPod and Modal seems to be defaulting to Arial (I'm viewing in FireFox)

- Designs display expertise first, then research focus below. Current Modal displays them the wrong way round.

**Design**

screen shot 2018-11-27 at 11 31 52

**Implementation**

screen shot 2018-11-27 at 11 34 03


- Can we keep the plural of 'Expertise' as 'Expertise' rather than 'Expertises' as currently? (e.g. Expertise: Neuroscience and Cell Biology)

- Space between Name and Institution is too large currently (see designs), should be 9px. I've just had a play with the inspector tool and looks like we can get closer to the intended result by removing all bottom margin and padding.

- Likewise space between the different sections doesn't match designs (see measurements above). Can we tweak the padding so that it does?

File: app/components/ui/atoms/PersonInfo.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    aff: 'Max Planck Institute, Germany',
+    focuses: [
+      'genetics of global gene expression',
+      'mouse models of human disease',
+      'haematology',
+    ],
+    expertises: ['Neuroscience', 'Cell Biology'],
+  }}
+/>
+```
```
Comment (damnhipster) [inline app/components/ui/atoms/PersonInfo.md]: This seems to be failing on the styleguide

File: app/components/ui/molecules/ModalDialog.md [hunk 2]
```diff
@@ -35,3 +40,40 @@ const FormH2 = require('../atoms/FormHeadings.js').FormH2
   </ModalDialog>
 </div>
 ```
+
+PersonInfo modal:
```
Comment (damnhipster) [inline app/components/ui/molecules/ModalDialog.md]: This is failing on the styleguide, for the same reason as the `PeopleInfo` failure

File: app/components/ui/atoms/PersonInfo.md (hunk 1 above)
Comment (damnhipster) [reply app/components/ui/atoms/PersonInfo.md]: screen shot 2018-11-27 at 12 14 56

File: app/components/pages/SubmissionWizard/steps/Editors/PeoplePickerControl.js [hunk 3]
```diff
@@ -19,13 +18,15 @@ const PeoplePickerControl = ({
...
+          focuses={person.focuses}
           iconType="remove"
           institution={person.aff}
           isKeywordClickable={false}
+          isSelected
           key={person.id}
-          keywords={person.subjectAreas}
           name={person.name}
-          onIconClick={() => onRequestRemove(person)}
+          togglePersonSelection={() => onRequestRemove(person)}
```
[the remark this answers is not in the document]
Comment (jsms90) [reply app/components/pages/SubmissionWizard/steps/Editors/PeoplePickerControl.js]: :grin:

File: app/components/ui/molecules/PeoplePicker.js [hunk 4]
```diff
@@ -70,16 +68,18 @@ const PeoplePickerBody = ({
...
               isSelected(person) || selection.length < maxSelection
             }
-            isKeywordClickable={false}
+            isSelected={isSelected(person)}
             key={person.id}
-            keywords={person.subjectAreas}
             name={person.name}
-            onIconClick={() => toggleSelection(person)}
+            togglePersonSelection={() => toggleSelection(person)}
             // onKeywordClick will need to be added, once we know what the desired behaviour is
```
[the remark this answers is not in the document]
Comment (jsms90) [reply app/components/ui/molecules/PeoplePicker.js]: How come? I think we want this in future.

**Comment by jsms90** [comment]
Thanks for the design review @hdrury1 :+1:

#### Font
I see why it feels appropriate to fix the font issue while we're touching this bit of the codebase, but the underlying reason why this problem exists has wider implications for the rest of the codebase. So I've raised a new issue #1122 & I would suggest that we resolve it as part of a separate PR. Is that ok with you?

#### Expertise vs expertises
Plural of "expertise" is "expertise" - mind blown :boom: :open_mouth: Cool.
I was intending to change this everywhere in the codebase, but I notice that we use "expertises" within MECA. So I've left it as "expertises" everywhere in our codebase except for the actual thing that's rendered to the user. Is that ok with you, @diversemix, in terms of consistency?

#### Spacing
I'm pretty sure that the latest changes reflect the designs. But I'm unsure whether I've missed something with this:
> Likewise space between the different sections doesn't match designs (see measurements above). Can we tweak the padding so that it does?

@hdrury1 I've removed the margin below the heading & added more margin below the focuses. Does that cover what you were thinking of here?

**Comment by jsms90** [comment]
@diversemix in what situation do you get this console error?

**Comment by diversemix** [comment]
Looks like its no longer an issue :+1:  so merging :)

**Pull request merged**